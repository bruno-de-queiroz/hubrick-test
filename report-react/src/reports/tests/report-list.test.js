import React from 'react';
import ReportList from '../report-list.component';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {configure, mount} from "enzyme";

configure({ adapter: new Adapter() });

it('should render the components', () => {
    const data = [{id: 'a', state: 'OPEN', payload: {reportId: 'bcada', reportType: 'SCAM', message: 'Test'}}];
    const component = renderer.create(<ReportList items={data} onBlock={() => {}} onResolve={() => {}}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('should filter out the closed tickets', () => {
    const data = [
        {id: 'a', state: 'OPEN', payload: {reportId: 'bcada', reportType: 'SCAM', message: 'Test'}},
        {id: 'b', state: 'CLOSED', payload: {reportId: 'cbada', reportType: 'SCAM', message: 'Test'}},
    ];
    const component = mount(<ReportList items={data} onBlock={() => {}} onResolve={() => {}}/>);
    expect(component.find('.report-item').length).toBe(1);
});