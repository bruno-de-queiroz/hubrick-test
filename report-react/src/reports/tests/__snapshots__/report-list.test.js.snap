// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the components 1`] = `
<ul
  className="report-list"
>
  <li
    className="report-item"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-10"
      >
        <div
          className="row"
        >
          <dl
            className="col-xs-5"
          >
            <dt>
              Id:
            </dt>
            <dd>
              bcada
            </dd>
          </dl>
          <dl
            className="col-xs-5"
          >
            <dt>
              Type:
            </dt>
            <dd>
              scam
            </dd>
          </dl>
        </div>
        <div
          className="row"
        >
          <dl
            className="col-xs-5"
          >
            <dt>
              State:
            </dt>
            <dd>
              open
            </dd>
          </dl>
          <dl
            className="col-xs-5"
          >
            <dt>
              Message:
            </dt>
            <dd>
              Test
            </dd>
          </dl>
        </div>
      </div>
      <div
        className="col-xs-2"
      >
        <button
          className="btn btn-warning"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Block
        </button>
        <button
          className="btn btn-success"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Resolve
        </button>
      </div>
    </div>
  </li>
</ul>
`;
